browser/widevine/widevine_permission_request.cc

```diff
@@ -33,31 +38,45 @@ WidevinePermissionRequest::WidevinePermissionRequest(
 
 WidevinePermissionRequest::~WidevinePermissionRequest() = default;
 
+#if BUILDFLAG(IS_ANDROID)
+std::u16string WidevinePermissionRequest::GetDialogMessageText() const {
+  return l10n_util::GetStringFUTF16(
+      GetWidevinePermissionRequestTextFrangmentResourceId(false),
+      url_formatter::FormatUrlForSecurityDisplay(
+          requesting_origin(),
+          url_formatter::SchemeDisplay::OMIT_CRYPTOGRAPHIC));
+}
+#else
 std::u16string WidevinePermissionRequest::GetMessageTextFragment() const {
   return brave_l10n::GetLocalizedResourceUTF16String(
       GetWidevinePermissionRequestTextFrangmentResourceId(for_restart_));
 }
+#endif
 
 void WidevinePermissionRequest::PermissionDecided(ContentSetting result,
                                                   bool is_one_time,
                                                   bool is_final_decision) {
   // Permission granted
   if (result == ContentSetting::CONTENT_SETTING_ALLOW) {
-#if BUILDFLAG(IS_LINUX)
-    // Prevent relaunch during the browser test.
-    // This will cause abnormal termination during the test.
-    if (for_restart_ && !is_test_) {
-      // Try relaunch after handling permission grant logics in this turn.
-      base::SequencedTaskRunner::GetCurrentDefault()->PostTask(
-          FROM_HERE, base::BindOnce(&chrome::AttemptRelaunch));
-    }
-#endif
     if (!for_restart_) {
-      EnableWidevineCdmComponent();
+      EnableWidevineCdm();
+    } else {
+#if BUILDFLAG(IS_ANDROID)
+      EnableWidevineCdm();
```

Comment:
why is `EnableWidevineCdm` called in the `else` here? Also android is always `!for_restart_`, right? How would you ever hit this IS_ANDROID block?

Reply:
No, Android always restarts, because the render client will not re-init with only a reload of page.

Reply:
On Linux `EnableWidevineCdm` shall not be called when restarts because it is already enabled in the first popup.

Reply:
wait, the renderer client does not have to re-init because BraveRendererUpdater will send the new pref value whenever it changes. There should be no reason for restart on android

Reply:
every frame load should trigger a call to `GetSupportedKeySystems`